### PR TITLE
checkcask 0.1.0 (new formula)

### PR DESCRIPTION
**Pull request by AcePeak**
Show every result of "brew cask list", and for each one of them, show the current remote latest version and the current local version.

- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by woodruffw** [comment]
Thanks for your submission. Unfortunately, we can't accept this as-is:

```
Error: 9 problems in 1 formula
checkcask:
  * Stable: version 0.1.0 is redundant with version scanned from URL
  * GitHub repository not notable enough (<20 forks, <20 watchers and <50 stars)
  * GitHub repository too new (<30 days old)
  * fully scope test system calls e.g. system "#{bin}/checkcask"
  * Please remove default template comments
  * Please remove default template comments
  * Please remove default template comments
  * C: 13: col 5: Missing space after #.
  * C: 17: col 5: Missing space after #.
```

We also generally prefer that packages be submitted by someone other than their primary maintainer - it gives us an indication that a community will be keeping the package alive.

In the mean time, you can always host this formula in a [personal tap](https://github.com/Homebrew/brew/blob/master/docs/How-to-Create-and-Maintain-a-Tap.md). I'll also leave some tips for reference during future submissions. Sorry, and thanks again!

File: Formula/checkcask.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Documentation: https://github.com/Homebrew/brew/blob/master/docs/Formula-Cookbook.md
```
Comment (woodruffw) [inline Formula/checkcask.rb]: These lines should be removed.

File: Formula/checkcask.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+#                http://www.rubydoc.info/github/Homebrew/brew/master/Formula
+# PLEASE REMOVE ALL GENERATED COMMENTS BEFORE SUBMITTING YOUR PULL REQUEST!
+
+class Checkcask < Formula
+  desc "Checks which cask is to update for brew cask."
+  homepage "https://github.com/AcePeak/checkcask"
+  url "https://github.com/AcePeak/checkcask/releases/download/v0.1.0/checkcask-0.1.0.tar.gz"
+  version "0.1.0"
```
Comment (woodruffw) [inline Formula/checkcask.rb]: This gets auto-detected from `url`.

File: Formula/checkcask.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  sha256 "b6eeccfce1df52ceb2b3b31318f09b6ee80639d9c8aaf8189d75648e105b3967"
+
+  def install
+    #system "./configure", "--disable-debug",
```
Comment (woodruffw) [inline Formula/checkcask.rb]: These lines should be removed.

File: Formula/checkcask.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    #                      "--disable-dependency-tracking",
+    #                      "--disable-silent-rules",
+    #                      "--prefix=#{prefix}"
+    #system "make", "install" # if this fails, try separate make/make install steps
+    bin.install "checkcask"
+  end
+
+  test do
+    # `test do` will create, run in and delete a temporary directory.
```
Comment (woodruffw) [inline Formula/checkcask.rb]: These lines should be removed.

File: Formula/checkcask.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    #
+    # This test will fail and we won't accept that! It's enough to just replace
+    # "false" with the main program this formula installs, but it'd be nice if you
+    # were more thorough. Run the test with `brew test checkcask`. Options passed
+    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
+    #
+    # The installed folder is not in the path, so use the entire path to any
+    # executables being tested: `system "#{bin}/program", "do", "something"`.
+    system "checkcask"
```
Comment (woodruffw) [inline Formula/checkcask.rb]: It's good to check the output of the commands you run against some expectation via `assert_match`.

**Comment by AcePeak** [comment]
Thanks William and sorry for my inexact submission.

I've fixed my codebase's defects and most problems below, except some "hard" conditions. Thanks for your tips, now I've setup my personal tap @https://github.com/AcePeak/homebrew-ace , and push this checkcask formula there. 

I can install it by following steps:

###
Brew tap AcePeak/homebrew-ace
Brew install checkcask
###

Hopefully it helps someone for cask uses. Thanks for correcting my submission, that would help me avoid future mistakes.

-Ace


Thanks for your submission. Unfortunately, we can't accept this as-is:
Error: 9 problems in 1 formula
checkcask:
  * Stable: version 0.1.0 is redundant with version scanned from URL
  * GitHub repository not notable enough (<20 forks, <20 watchers and <50 stars)
  * GitHub repository too new (<30 days old)
  * fully scope test system calls e.g. system "#{bin}/checkcask"
  * Please remove default template comments
  * Please remove default template comments
  * Please remove default template comments
  * C: 13: col 5: Missing space after #.
  * C: 17: col 5: Missing space after #.
We also generally prefer that packages be submitted by someone other than their primary maintainer - it gives us an indication that a community will be keeping the package alive.

In the mean time, you can always host this formula in a personal tap. I'll also leave some tips for reference during future submissions. Sorry, and thanks again!

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.

  
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/Homebrew/homebrew-core","title":"Homebrew/homebrew-core","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/Homebrew/homebrew-core"}},"updates":{"snippets":[{"icon":"PERSON","message":"@woodruffw in #9152: Thanks for your submission. Unfortunately, we can't accept this as-is:\r\n\r\n```\r\nError: 9 problems in 1 formula\r\ncheckcask:\r\n  * Stable: version 0.1.0 is redundant with version scanned from URL\r\n  * GitHub repository not notable enough (\u003c20 forks, \u003c20 watchers and \u003c50 stars)\r\n  * GitHub repository too new (\u003c30 days old)\r\n  * fully scope test system calls e.g. system \"#{bin}/checkcask\"\r\n  * Please remove default template comments\r\n  * Please remove default template comments\r\n  * Please remove default template comments\r\n  * C: 13: col 5: Missing space after #.\r\n  * C: 17: col 5: Missing space after #.\r\n```\r\n\r\nWe also generally prefer that packages be submitted by someone other than their primary maintainer - it gives us an indication that a community will be keeping the package alive.\r\n\r\nIn the mean time, you can always host this formula in a [personal tap](https://github.com/Homebrew/brew/blob/master/docs/How-to-Create-and-Maintain-a-Tap.md). I'll also leave some tips for reference during future submissions. Sorry, and thanks again!"}],"action":{"name":"View Pull Request","url":"https://github.com/Homebrew/homebrew-core/pull/9152#issuecomment-274346875"}}}

**Comment by AcePeak** [comment]
Thanks William for your advice.

For the test section, I thought for some time but had no solution for such case. 

"checkcask" is a extreme simple shell command that output all cask apps' remote version and local version to diff, , the results look like:


generally it recommends user to open a new terminal to run "brew cask reinstall [app]" according to user's demand. PS: By my 

That's the only function. So I just don't know how to test it by using some assertions. I do need helps and advices for this part for checkcask formula.

Could you please help me to point out a way for this case? Thanks so much!

-Ace


@woodruffw commented on this pull request.

In Formula/checkcask.rb:
> +    #                      "--prefix=#{prefix}"
+    #system "make", "install" # if this fails, try separate make/make install steps
+    bin.install "checkcask"
+  end
+
+  test do
+    # `test do` will create, run in and delete a temporary directory.
+    #
+    # This test will fail and we won't accept that! It's enough to just replace
+    # "false" with the main program this formula installs, but it'd be nice if you
+    # were more thorough. Run the test with `brew test checkcask`. Options passed
+    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
+    #
+    # The installed folder is not in the path, so use the entire path to any
+    # executables being tested: `system "#{bin}/program", "do", "something"`.
+    system "checkcask"
It's good to check the output of the commands you run against some expectation via assert_match.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.

  
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/Homebrew/homebrew-core","title":"Homebrew/homebrew-core","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/Homebrew/homebrew-core"}},"updates":{"snippets":[{"icon":"PERSON","message":"@woodruffw commented on #9152"}],"action":{"name":"View Pull Request","url":"https://github.com/Homebrew/homebrew-core/pull/9152#pullrequestreview-17833887"}}}

**Comment by woodruffw** [comment]
@AcePeak It's fine if there's no way to test your formula in a more in-depth way. We just have to ask because we aren't always familiar with the new package.

**Pull request closed without merging**